agr_literature_service/api/crud/reference_crud.py

```diff
@@ -889,102 +934,148 @@ def get_bib_info(db, curie, mod_abbreviation: str, return_format: str = 'txt'):
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
-                            detail="The reference_type passed in is not an valid reference_type.")
+                            detail="The reference_type passed in is not a valid reference_type.")
+
     mod = db.query(ModModel).filter_by(abbreviation=mod_abbreviation).one_or_none()
     if not mod:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
                             detail=f"The mod_abbreviation: {mod_abbreviation} is not in the database.")
 
+    mod_id = mod.mod_id
+
     """
     Caenorhabditis elegans   NCBITaxon:6239
     Caenorhabditis brenneri  NCBITaxon:135651
     Caenorhabditis briggsae  NCBITaxon:6238
     Caenorhabditis japonica  NCBITaxon:281687
     Caenorhabditis remanei   NCBITaxon:31234
-    Brugia malayi 	     NCBITaxon:6279
+    Brugia malayi            NCBITaxon:6279
     Onchocerca volvulus      NCBITaxon:6282
     Pristionchus pacificus   NCBITaxon:54126
     Strongyloides ratti      NCBITaxon:34506
     Trichuris muris          NCBITaxon:70415
     """
 
-    wb_textpresso_species_list = "'NCBITaxon:6239', 'NCBITaxon:135651', 'NCBITaxon:6238', 'NCBITaxon:281687', 'NCBITaxon:31234', 'NCBITaxon:6279', 'NCBITaxon:6282', 'NCBITaxon:54126', 'NCBITaxon:34506', 'NCBITaxon:70415'"
+    wb_textpresso_species_list = (
+        'NCBITaxon:6239', 'NCBITaxon:135651', 'NCBITaxon:6238', 'NCBITaxon:281687',
+        'NCBITaxon:31234', 'NCBITaxon:6279', 'NCBITaxon:6282', 'NCBITaxon:54126',
+        'NCBITaxon:34506', 'NCBITaxon:70415'
+    )
+
+    # Start building the query string
+    query_str = """
+        SELECT r.curie, r.reference_id, rf.referencefile_id, rf.md5sum, rfm.mod_id, rf.date_created
+        FROM reference r
+        JOIN mod_corpus_association mca on r.reference_id = mca.reference_id
+        JOIN referencefile rf ON rf.reference_id = r.reference_id
+        JOIN referencefile_mod rfm ON rf.referencefile_id = rfm.referencefile_id
+        WHERE mca.corpus is True
+        AND mca.mod_id = :mod_id
+        AND rf.file_class = 'main'
+        AND rf.file_extension = 'pdf'
+        AND (rfm.mod_id is NULL OR rfm.mod_id = :mod_id)
+    """
 
-    mod_id = mod.mod_id
-    select_stmt = f"""SELECT r.curie, r.reference_id, rf.referencefile_id, rf.md5sum, rfm.mod_id, rf.date_created
-      FROM reference r
-      JOIN mod_corpus_association mca on r.reference_id = mca.reference_id
-      JOIN referencefile rf ON rf.reference_id = r.reference_id
-      JOIN referencefile_mod rfm ON rf.referencefile_id = rfm.referencefile_id
-      WHERE mca.corpus is True
-      AND mca.mod_id = {mod_id}
-      AND rf.file_class = 'main'
-      AND rf.file_extension = 'pdf'
-      AND (rfm.mod_id is NULL OR rfm.mod_id = {mod_id})"""
+    query_params = {'mod_id': mod_id}
 
+    # Add condition for reference_type if provided
     if reference_type:
-        select_stmt += f"""
+        query_str += """
         AND r.reference_id IN (
```

Comment:
Did sqlalchemy not like the old syntax and wanting JOIN, or was it something else ?  I recall it didn't like {mod_id} and wanting :mod_id  but does :mod_id not work without JOIN or something like that ?

Reply:
Good question! @azurebrd  The old syntax is still working for version 2.0.7 of SQLAlchemy.
Just JOIN makes things clear (at least to me). The :mod_id should work both with and
without a JOIN. The {mod_id} syntax is still working in this version (2.0.7), but it may have
potential security concerns although in our case it should not be a big deal (since we are
using the query to select the data from our database). Anyway, github security check wasn't
happy about it and the SQLAlchemy 2.x does enforce stricter SQL parameterization rules, and
the :mod_id syntax meets the SQL standard of binding parameter requirement for the SQLAlchemy 2.x.

Reply:
Cool cool, good to know, thanks !